### DISCUSSION TITLE
MQTT 5 implement

### DISCUSSION THREAD

**Discussion by oceanlvr** [emqx/MQTTX#744] [Ideas]
We plan to support `MQTT 5`, here discussion used for detail design.  🤔 🤔 🤔 

All feedback is welcome! ⚡ ⚡

**Comment by aaron-613** [comment]
Hello!  IMHO, the most important update from MQTT5 that would be relevant to MQTTX would be the ability to see (and possibly set) the custom **User Properties** that can now be passed with messages.  Ideally, they would be displayed inside the message bubbles when sending/receiving, but hopefully at least these would be visible in the logs or something.

Other than that, showing the Reason Codes in the log when getting a negative response to a action/command would be great!

**Reply by ysfscream** [reply]
We are currently working on this feature, but do you have any good ideas for the UI in terms of setting and displaying User Properties? Maybe your ideas will become reality in the next release! 😝

**Comment by ysfscream** [comment]
Hey! Now we support User Properties on https://github.com/emqx/MQTTX/releases/tag/v1.7.0. Although there still seems to be a little problem but should be able to use it!
